src/plugins/Dropbox/index.js

```diff
@@ -0,0 +1,128 @@
+import Plugin from '../Plugin'
+import 'whatwg-fetch'
+import html from '../../core/html'
+
+import Provider from '../../uppy-base/src/plugins/Provider'
+
+import View from '../../generic-provider-views/index'
+import icons from './icons'
+
+export default class Dropbox extends Plugin {
+  constructor (core, opts) {
+    super(core, opts)
+    this.type = 'acquirer'
+    this.id = 'Dropbox'
+    this.title = 'Dropbox'
+    this.stateId = 'dropbox'
+    this.icon = html`
+      <svg class="UppyIcon UppyModalTab-icon" viewBox="0 0 225.000000 225.000000">
+        <g transform="translate(0.000000,225.000000) scale(0.100000,-0.100000)" fill="#000000" stroke="none">
+        <path d="M633 1646 c-112 -72 -203 -134 -203 -137 0 -3 61 -54 135 -113 74 -59 135 -111 135 -116 0 -5 -61 -57
+        -135 -116 -74 -59 -135 -110 -135 -113 1 -7 391 -261 403 -262 5 -1 72 52 149 116 l141 118 -54 33 c-30 18 -123 75
+        -206 126 -84 51 -153 95 -153 98 0 3 69 47 153 98 83 51 176 108 206 126 l53 33 -123 103 c-68 57 -133 111 -144
+        121 -19 16 -27 12 -222 -115z"/>
+        <path d="M1271 1657 l-143 -120 53 -33 c30 -18 123 -75 206 -126 84 -51 153 -95 153 -98 0 -3 -69 -47 -153 -98 -83
+        -51 -176 -108 -206 -126 l-54 -33 139 -116 c77 -64 142 -117 145 -119 8 -3 408 254 409 263 0 3 -61 54 -135 113 -74
+        59 -135 111 -135 116 0 5 61 57 135 116 74 59 135 110 135 113 0 3 -51 39 -113 78 -61 40 -153 99 -202 132 l-91 58
+        -143 -120z"/>
+        <path d="M981 849 c-159 -132 -141 -127 -228 -67 l-43 29 0 -44 0 -45 207 -123 c114 -68 210 -123 213 -122 4 1 98 57
+        209 123 l201 121 0 45 0 45 -61 -41 c-52 -36 -62 -40 -77 -28 -9 7 -69 56 -132 108 -63 52 -122 100 -130 107
+        -13 10 -40 -9 -159 -108z"/>
+        </g>
+      </svg>`
+
+    this[this.id] = new Provider({
```

Comment:
I’m not sure, you are right about “the generic view makes use of it that way”. Maybe lets leave it for now, setting id in one place and not repeating ourselves might be a good idea.